File: SampleApp/Objective-C/SampleApp/Banner/ManyBannerViewController.m
//
//  ManyBannerViewController.m
//  SampleApp
//
//  Fluct SDK
//  Copyright (c) 2020 fluct, Inc. All rights reserved.
//

#import "ManyBannerViewController.h"
#import <FluctSDK/FluctSDK.h>

@interface ManyBannerViewController ()
@property (nonatomic) FSSAdView *topAdView;
@property (nonatomic) FSSAdView *bottomAdView;
@end

@implementation ManyBannerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    FSSAdView *topAdView = [[FSSAdView alloc] initWithGroupId:@"1000055927" unitId:@"1000084701" adSize:FSSAdSize320x50];
    [self.view addSubview:topAdView];
    [topAdView loadAd];
    self.topAdView = topAdView;

    FSSAdView *bottomAdView = [[FSSAdView alloc] initWithGroupId:@"1000055927" unitId:@"1000084701" adSize:FSSAdSize320x50];
    [self.view addSubview:bottomAdView];
    [bottomAdView loadAd];
    self.bottomAdView = bottomAdView;
}

- (void)viewDidLayoutSubviews {
    [super viewDidLayoutSubviews];
    CGFloat minY = CGRectGetMinY(self.view.bounds);
    CGFloat topAdViewY = minY + self.view.layoutMargins.top;

    CGFloat maxY = CGRectGetMaxY(self.view.bounds);
    CGFloat bottomAdViewY = maxY - self.view.layoutMargins.bottom - CGRectGetHeight(self.bottomAdView.frame);

    CGFloat midX = CGRectGetMidX(self.view.bounds);
    CGFloat topAdViewX = midX - CGRectGetWidth(self.topAdView.frame) * 0.5;
    CGFloat bottomAdViewX = midX - CGRectGetWidth(self.bottomAdView.frame) * 0.5;

    CGRect topAdViewFrame = self.topAdView.frame;
    topAdViewFrame.origin = CGPointMake(topAdViewX, topAdViewY);
    self.topAdView.frame = topAdViewFrame;

    CGRect bottomAdViewFrame = self.bottomAdView.frame;
    bottomAdViewFrame.origin = CGPointMake(bottomAdViewX, bottomAdViewY);
    self.bottomAdView.frame = bottomAdViewFrame;
}

@end

File: SampleApp/Swift/SDK/SampleApp/Banner/ManyBannerViewController.swift
//
//  ManyBannerViewController.swift
//  SampleApp
//
//  Fluct SDK
//  Copyright © 2016年 fluct, Inc. All rights reserved.
//

import UIKit
import FluctSDK

class ManyBannerViewController: UIViewController {

    private var topAdView: FSSAdView?
    private var bottomAdView: FSSAdView?

    override func viewDidLoad() {
        super.viewDidLoad()

        let topAdView = FSSAdView(groupId: "1000055927", unitId: "1000084701", adSize: FSSAdSize320x50)
        self.view.addSubview(topAdView)
        topAdView.loadAd()
        self.topAdView = topAdView

        let bottomAdView = FSSAdView(groupId: "1000055927", unitId: "1000084701", adSize: FSSAdSize320x50)
        self.view.addSubview(bottomAdView)
        bottomAdView.loadAd()
        self.bottomAdView = bottomAdView
    }

    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()

        let minY = self.view.bounds.minY
        let topAdViewY = minY + self.view.layoutMargins.top

        let maxY = self.view.bounds.maxY
        let bottomAdViewHeight = self.bottomAdView?.frame.height ?? 0.0
        let bottomAdViewY = maxY - self.view.layoutMargins.bottom - bottomAdViewHeight

        let midX = self.view.bounds.midX
        let topAdViewWidth = self.topAdView?.frame.width ?? 0.0
        let topAdViewX = midX - topAdViewWidth * 0.5

        let bottomAdViewWidth = self.bottomAdView?.frame.width ?? 0.0
        let bottomAdViewX = midX - bottomAdViewWidth * 0.5

        var topAdViewFrame = self.topAdView?.frame ?? .zero
        topAdViewFrame.origin = CGPoint(x: topAdViewX, y: topAdViewY)
        self.topAdView?.frame = topAdViewFrame

        var bottomAdViewFrame = self.bottomAdView?.frame ?? .zero
        bottomAdViewFrame.origin = CGPoint(x: bottomAdViewX, y: bottomAdViewY)
        self.bottomAdView?.frame = bottomAdViewFrame

    }
}
